pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/services/IdMinterWorkerService.scala

```diff
@@ -33,12 +41,37 @@ class IdMinterWorkerService[Destination](
       tableName = identifiersTableConfig.tableName
     )
 
-    messageStream.foreach(this.getClass.getSimpleName, processMessage)
+    messageStream.runStream(
+      className,
+      _.via(extractIdentifiers)
+        .via(synchroniseIds)
+        .via(embedIdentifiers)
+        .via(sendMinted)
+        .map { case (msg, _) => msg }
+    )
   }
 
-  def processMessage(json: Json): Future[Unit] =
-    for {
-      identifiedJson <- idEmbedder.embedId(json)
-      _ <- Future.fromTry { sender.sendT(identifiedJson) }
-    } yield ()
+  private def extractIdentifiers =
+    Flow[(Message, Json)].map {
+      case (msg, json) =>
+        (msg, json, SourceIdentifierScanner.scan(json).get)
```

Comment:
Why are we doing a `.get` here? I thought that was bad practice. (Ditto below.)

Reply:
Because if there's an exception we want to throw it so akka-streams can do it's thing.

I was going to move all of the exception logging to here, but then I found that there's so much context that's useful to have (as you've noticed in other comments) in the logs that didn't work very well. 

When this was a for comprehension (which you can see in the commit history) it was all rather nice but it was necessary to move to streams to deal with concurrency issues.

I might add an error-handling convenience method to tack on the end of each flow and strip off the `Try` instead, what do you think?